operator/operatortest/envtest.go

```diff
@@ -181,6 +181,31 @@ func (ts *Suite) FetchResources(objectList client.ObjectList, opts ...client.Lis
 	ts.Require().NoError(ts.Client.List(ts.Context, objectList, opts...))
 }
 
+// ExistsResource checks if the given resource exists.
+// Test fails for any error.
+func (ts *Suite) ExistsResource(name types.NamespacedName, object client.Object) {
+	ts.Require().NoError(ts.Client.Get(ts.Context, name, object))
+}
+
+// NotFoundResource checks if the given resource is missing.
+// Test fails if the resource exists or there's another error.
+func (ts *Suite) NotFoundResource(name types.NamespacedName, object client.Object) {
+	err := ts.Client.Get(ts.Context, name, object)
+	if err != nil {
+		ts.Require().True(apierrors.IsNotFound(err))
+	} else {
+		ts.Require().NotEmpty(object.GetDeletionTimestamp())
+	}
+}
```

Comment:
Several things:
- We can omit the NamespacedName from the signature by replacing it `client.ObjectKeyFromObject`, provided the object has Name and Namespace set in the Object meta.
- When doing a Get, the given object will be overwritten with the contents. This is unacceptable for an assert function. Thus we need to DeepCopy the current object so that the original object is not modified.
- I would not use `NotEmpty`, AFAIR this is used for maps and slices. To check for nil use `Nil`/`NotNil`. But AFAIR the deletionTimestamp is of a special Time struct that has `IsZero()` and this might be better suited for testing as it can also handle nil timestamps.
```suggestion
// AssertResourceNotExists checks if the given resource is not existing or is existing with a deletion timestamp.
// Test fails if the resource exists or there's another error.
func (ts *Suite) AssertResourceNotExists(object client.Object) {
	err := ts.Client.Get(ts.Context, client.ObjectKeyFromObject(object), object.DeepCopy())
	if err != nil {
		ts.Require().True(apierrors.IsNotFound(err))
	} else {
		ts.Require().True(object.GetDeletionTimestamp().IsZero())
	}
}
```

Reply:
Thanks for clearing out. Point 2 was an interesting finding. Indeed if the method is used to pass the whole object then it will override it. In my case I used it exclusively with empty object.